contracts/SpokePool.sol

```diff
@@ -147,19 +187,73 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
     }
 
     function initiateRelay(
-        uint256 originChain,
-        address sender,
+        uint256 originChainId,
         uint256 amount,
-        address recipient,
-        uint256 relayerFee,
-        uint256 realizedLpFee
-    ) public {}
+        uint64 depositId,
+        uint64 relayerFeePct,
+        uint64 realizedLpFeePct,
+        address destinationToken,
+        address sender,
+        address recipient
+    ) public {
+        // We limit the relay fees to prevent the user spending all their funds on fees.
+        require(relayerFeePct <= 0.5e18 && realizedLpFeePct <= 0.5e18, "invalid fees");
+
+        // Relay will be keyed to unique deposit plus relay data hash.
+        RelayData memory relayData = RelayData(
```

Comment:
is it cheaper to get the relayer to pass a `memory relayData` as a prop in this function and not construct this in the method like this?

Reply:
main downside is this makes calling this method a bit more complex and we would require the caller to set the `totalFillAmount` to 0.

Reply:
if you choose to leave it like this, can you use Key-value assignment syntax that we use here: https://github.com/across-protocol/across-smart-contracts-v2/blob/977a9a48f8b3993c95d2a4d724462355823c2458/contracts/HubPool.sol?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L82

Reply:
ended up using key-value syntax but not going to make caller pass in `relayData` because it marginally makes the interface harder to use for EOA's on etherscan

Reply:
it adds a little calldata too

Reply:
+1 @nicholaspai